process/dns.go

```diff
@@ -0,0 +1,188 @@
+package process
+
+import (
+	"context"
+	"fmt"
+	"net"
+	"time"
+
+	"github.com/brexhq/substation/condition"
+	"github.com/brexhq/substation/config"
+)
+
+var dnsResolver net.Resolver
+
+/*
+DNS processes data by querying domains or IP addresses in the Domain Name System (DNS). By default, this processor can take up to 1 second per DNS query and may have significant impact on end-to-end data processing latency. If Substation is running in AWS Lambda with Kinesis, then this latency can be mitigated by increasing the parallelization factor of the Lambda (https://docs.aws.amazon.com/lambda/latest/dg/with-kinesis.html).
+
+The processor supports these patterns:
+
+	JSON:
+	  	{"ip":"8.8.8.8"} >>> {"ip":"8.8.8.8","domains":["dns.google."]}
+		{"domain":"dns.google"} >>> {"domain":"dns.google","ips":["8.8.4.4","8.8.8.8","2001:4860:4860::8844","2001:4860:4860::8888"]}
+	data:
+		8.8.8.8 >>> dns.google.
+		dns.google >>> 8.8.4.4
+
+When loaded with a factory, the processor uses this JSON configuration:
+
+	{
+		"type": "dns",
+		"settings": {
+			"options": {
+				"function": "reverse_lookup"
+			},
+			"input_key": "ip",
+			"output_key": "domains"
+		}
+	}
+*/
+type DNS struct {
```

Comment:
test: This processor doesn't have any tests added, is this intentional due to the network processing that tests would require?

Reply:
yep, it is intentional. i added [examples](https://go.dev/blog/examples) to provide some test coverage, though they aren't executed during unit testing. i'm open to suggestions on how to do actual unit tests for this though.